src/NzbDrone.Mono/Disk/DiskProvider.cs

```diff
@@ -189,6 +194,46 @@ public override bool TryCreateHardLink(string source, string destination)
             }
         }
 
+        public override bool TryCreateRefLink(string source, string destination)
+        {
+            var source_fd = Syscall.open(source, OpenFlags.O_RDONLY);
+            if (source_fd == -1) {
+                Logger.Trace("RefLink: Cannot open source file '{0}'", source);
+                return false;
+            }
+
+            var dest_fd = Syscall.open(destination, OpenFlags.O_WRONLY | OpenFlags.O_CREAT, FilePermissions.S_IRUSR | FilePermissions.S_IWUSR);
+            if (dest_fd == -1) {
+                Syscall.close(source_fd);
+                Logger.Trace("RefLink: Cannot open dest file '{0}'", destination);
+                return false;
+            }
+
+            try
+            {
+                var result = FiCloneIoCtl(dest_fd, FICLONE, source_fd);
+                if (result == 0) {
+                    Syscall.close(source_fd);
+                    Syscall.close(dest_fd);
+                    return true;
+                }
+                var error = Stdlib.GetLastError();
+                Logger.Debug("Cannot issue FICLONE ('{0}' -> '{1}') due to '{2}'", Stdlib.strerror(error));
+            }
+            catch (EntryPointNotFoundException e)
```

Comment:
Can this throw any other types of exception that won't get caught and mean that the two file descriptors don't get closed?

Reply:
https://stackoverflow.com/questions/421797/what-really-happens-in-a-try-return-x-finally-x-null-statement seems to tell me I should close these things in a `finally`. Added.